kotlin/src/test/Utils.kt

```diff
@@ -34,10 +34,14 @@ typealias jsonDict = Map<String, Any>
 
 val jsonDictType = object : TypeToken<jsonDict>() {}.type
 
-class TestSettingsIniFile(filename: String = "./looker.ini", section: String = "")
-    : ApiSettingsIniFile(filename, section) {
+class TestSettingsIniFile(
+    filename: String = "./looker.ini",
+    section: String = "",
+    private val base: ConfigurationProvider = ApiSettings.fromIniFile(filename, section)
+) : ConfigurationProvider by base {
+
     override fun readConfig(): Map<String, String> {
-        var map = super.readConfig().toMutableMap()
+        var map = base.readConfig().toMutableMap()
```

Comment:
does `base` go to the deepest ancestor rather than whatever's closest in the inheritance tree?

Reply:
It actually refers to `private val base: ConfigurationProvider` in the constructor.  I'm using kotlin's interface delegation here, since I want to "inherit" from ApiSettings, but I don't want to directly call its constructor.  Sticking that in the TestSettingsIniFile constructor feels like a bit of a hack, but I needed to be able to reference the delegated instance from somewhere.

https://kotlinlang.org/docs/reference/delegation.html

Reply:
ah, that's where it is! I read that same link but didn't see a ref to `base`. I missed the variable assignment. Sorry about that.

Reply:
No problem.  It's a bit too subtle for my taste, but I couldn't think of a better way to get what I wanted here.  In the Mock settings, we don't need to access anything from the base instance, and that came out a bit cleaner IMO: https://github.com/looker-open-source/sdk-codegen/pull/293/files#diff-40b1d00a327afcf5a64170b78a2de656R21